blog/tanstack-router-loaders/index.md

```diff
@@ -0,0 +1,630 @@
+
+You're probably wondering if TanStack Router tells you this background re-fetch is happening, so you can show an inline spinner, and yes, you can detect this via
+
+```ts
+const { isFetching } = Route.useMatch();
+```
+
+### Loader in a page
+
+Now let's take a look at the tasks page
+
+```ts
+export const Route = createFileRoute("/app/tasks/")({
+  component: Index,
+  loader: async ({ context }) => {
+    const now = +new Date();
+    console.log(`/tasks/index path loader. Loading tasks at + ${now - context.timestarted}ms since start`);
+
+    const tasks = await fetchJson<Task[]>("api/tasks");
+    return { tasks };
+  },
+  gcTime: 1000 * 60 * 5,
+  staleTime: 1000 * 60 * 2,
+  pendingComponent: () => <div className="m-4 p-4 text-xl">Loading tasks list...</div>,
+  pendingMs: 150,
+  pendingMinMs: 200,
+});
+```
+
+This is the route for the specific url `/app/tasks`. If the user were to browse to `/app/tasks/$taskId` then this component would not run. This is a specific page, not a layout (which Router calls a "route"). Basically the same as before, except now we're loading the list of tasks to display on this page.
+
+We've added some new properties this time though. `pendingComponent` allows us to render some content while the loader is working. We also specified `pendingMs`, which controls how long we _wait_ before showing the pending component. Lastly, `pendingMinMs` allows us to force the pending component to stay on the screen for a specified amount of time, even if the data are ready. This can be useful to avoid an extremely brief flash of a loading component, which can be jarring to the user.
+
+If you're wondering why we'd even want to use `pendingMs` to delay a loading screen, it's for subsequent navigations. Rather than _immediately_ transition from the current page to a new page's loading component, this setting lets us stay on the current page for a moment, in the hopes that the new page will be ready quickly enough that we don't have to show any pending component at all. Of course, on the initial load, when the web app first starts up, these pendingComponents do show immediately, as you'd expect.
+
+Let's run our tasks page
+
+![loaders running in parallel](/tanstack-router-loaders/img-0-tasks-page.jpg)
+
+It's ugly and frankly useless, but it works. Now let's take a closer look.
+
+### Loaders running in parallel
+
+If we peak in our dev tools, we should see something like this
+
+![loaders running in parallel](/tanstack-router-loaders/img-1-loaders-parallel.jpg)
+
+As we can see, these requests started a mere millisecond apart from each other, since the loaders are running in parallel (each request takes at least 750ms, due to the artificial delays in the api endpoints).
```

Comment:
> (each request takes at least 750ms, due to the artificial delays in the api endpoints).

I miss the witty criticism of jira being utterly slow :)

Reply:
I added it back just for you :)